### PR TITLE
[ONNX] Support None in fx.args as torchlib inputs

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108708

Prior to this PR, if None is returned from intermediate nodes, it will crashes the export because None is not expected to be passed into `_fill_tensor_shape_type`, and raise beartype roar. The function fills in shape and type to TorchScriptTensor according to its info from FX graph.

This is discovered after https://github.com/microsoft/onnxscript/pull/1043 is supported. The op specifically generates None in one of its inputs, but the only output from it being consumed is the first one (not None).

Reference test from a TorchBench model:
```python

    def test_nanogpt(self):
        import sys

        sys.path.append("/home/titaiwang")

        from nanoGPT.model import GPT, GPTConfig

        # Load the model
        kwargs = {
            "block_size": 256,
            "vocab_size": 8096,  # GPT-2 vocab_size of 50257, padded up to nearest multiple of 64 for efficiency
            "n_layer": 2,
            "n_head": 2,
            "n_embd": 128,
            "dropout": 0.0,
            "bias": False,  # True: bias in Linears and LayerNorms, like GPT-2. False: a bit better and faster
        }
        config = GPTConfig(**kwargs)
        with torch.backends.cuda.sdp_kernel(
            enable_flash=True, enable_mem_efficient=True
        ):
            model = GPT(config)
        print("Done loading model")
        inputs = torch.arange(128).view(2, 64)
        targets = torch.arange(128).view(2, 64)

        self.run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
            model,
            (inputs,),
            input_kwargs={
                "targets": targets,
            },
            verbose=True,
        )
```

### CODE REVIEW

**Comment by justinchuby** [comment]
> The op specifically generates None in one of its inputs, but the only output from it being consumed is the first one (not None).

Do you mean an input or output is None? If output which one?

File: torch/onnx/_internal/fx/fx_onnx_interpreter.py [hunk 1]
```diff
@@ -234,10 +234,12 @@ def _fill_tensor_shape_type(
     for i, (onnxscript_value, expected_value) in enumerate(
         zip(flat_onnxscript_values, flat_expected_values)
     ):
-        # aten::sym_size output is a int, not a tensor, which stands
-        # for the size of one dim. We treat it as 0-D tensor.
-        # TODO(titaiwang): set shape?
-        if isinstance(expected_value, (torch.SymInt, torch.SymFloat, torch.SymBool)):
+        if expected_value is None:
+            # There is no shape/type from None.
```
Comment (justinchuby) [inline torch/onnx/_internal/fx/fx_onnx_interpreter.py:238]: I would link to the example you shown so readers know when this will happen

Comment (justinchuby) [reply torch/onnx/_internal/fx/fx_onnx_interpreter.py:238]: Would it be possible for use to assume it is always a scalar? Or could there be other cases?

**Review by justinchuby — APPROVED** [review]

**Review by thiagocrepaldi — APPROVED** [review]

File: torch/onnx/_internal/fx/fx_onnx_interpreter.py (hunk 1 above)
Comment (titaiwangms) [reply torch/onnx/_internal/fx/fx_onnx_interpreter.py:238]: hmmm it all depends on what fx.node gives us. This is pure product from fx graph. So I feel like what we should is only taking `None` into consideration of our code.

Comment (titaiwangms) [reply torch/onnx/_internal/fx/fx_onnx_interpreter.py:238]: My guess is that for CPU, those outputs are useless or non-generated, so it returns None.

**Pull request closed without merging**